src/libraries/Microsoft.Extensions.Options/src/ValidateOptions.cs

```diff
@@ -27,30 +27,30 @@ public ValidateOptions(string name, Func<TOptions, bool> validation, string fail
         /// <summary>
         /// The options name.
         /// </summary>
-        public string Name { get; }
+        public string? Name { get; }
 
         /// <summary>
         /// The validation function.
         /// </summary>
-        public Func<TOptions, bool> Validation { get; }
+        public Func<TOptions, bool>? Validation { get; }
 
         /// <summary>
         /// The error to return when validation fails.
         /// </summary>
-        public string FailureMessage { get; }
+        public string? FailureMessage { get; }
 
         /// <summary>
         /// Validates a specific named options instance (or all when <paramref name="name"/> is null).
         /// </summary>
         /// <param name="name">The name of the options instance being validated.</param>
         /// <param name="options">The options instance.</param>
         /// <returns>The <see cref="ValidateOptionsResult"/> result.</returns>
-        public ValidateOptionsResult Validate(string name, TOptions options)
+        public ValidateOptionsResult Validate(string? name, TOptions options)
         {
             // null name is used to configure all named options
             if (Name == null || name == Name)
             {
-                if ((Validation?.Invoke(options)).Value)
+                if ((Validation?.Invoke(options))!.Value)
```

Comment:
This doesn't seem safe. If Validation is null, won't this null-ref?

repeated elsewhere

Reply:
Tested it, it will NRE. 
![image](https://user-images.githubusercontent.com/6609929/150652113-b4886bd9-6c58-4bb6-8e42-fd5dff6ff169.png)

I thought since we have `Nullable<t>` struct, it won't actually be `null` and just revert to `default` if there's no value or something. (still don't understand where `Nullable<t>` comes from though)

What should be done here? Can I change it to `if (Validation == null || Validation.Invoke(options))` in this PR?

Reply:
Maybe the best thing to do here is to make `Validation` be non-nullable, and start throwing in the ctor. Looking at the only callsite of it, I believe that was the original intention:

https://github.com/dotnet/runtime/blob/3c43ef6af7dd0a54e26ae23e4610f4937a7a7e4d/src/libraries/Microsoft.Extensions.Options/src/OptionsBuilder.cs#L368-L373

Reply:
The other option is to return `Skip` when the `Validation` function is `null`. I don't think we should return `Pass` or `Failed` - we really have no idea.

@HaoK - thoughts here?

Reply:
> make Validation be non-nullable, and start throwing in the ctor

Yeah that was the original intent, there shouldn't be null validation functions